mitodl/micromasters
# Added API to create Order and populate CyberSource parameters

#### What are the relevant tickets?
Fixes #891 

#### What's this PR do?
Adds `/api/v0/checkout/` to create a new `Order` and provide parameters for CyberSource

#### Where should the reviewer start?

#### How should this be manually tested?
 - Create a verified course on your devstack.
 - Set up a `CourseRun` object which matches the course on your devstack. Set the `start_date` to be before today and the other three date fields should be `None`. You should see an 'Enroll' button on the dashboard.
 - Create a `CoursePrice` object which points to the `CourseRun`. `is_valid` must be true, choose any positive value for the price.
 - Fill in the new environment variables in `.env`. Talk to me in private for the keys needed here
 - Click `Enroll`. You should be directed to a CyberSource page.
 - Enter fake information for the address component. For the credit card, use `4111111111111111`, the card type is `Visa`, the CVN is any three digits, and the expiration month is some time in the future. Enter a legit email address so you can see the receipt generated.
 - Pay for the course. You should be redirected to a request bin with the word `ok`.
 - Ask me for the purchase data email. It should report a successful purchase and include all the information entered (but only the last 4 digits of the credit card).

This workflow should also be the same for course runs with the button `Upgrade`.

#### Any background context you want to provide?

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
